File: app/src/main/java/me/devsaki/hentoid/activities/sources/BabeTodayActivity.java
package me.devsaki.hentoid.activities.sources;

import me.devsaki.hentoid.enums.Site;

public class BabeTodayActivity extends BaseWebActivity {

    private static final String DOMAIN_FILTER = "babe.today";
    private static final String[] GALLERY_FILTER = {"https://babe.today/.+$"};

    Site getStartSite() {
        return Site.BABETODAY;
    }

    @Override
    boolean allowMixedContent() {
        return false;
    }


    @Override
    protected CustomWebViewClient getWebClient() {
        CustomWebViewClient client = new CustomWebViewClient(GALLERY_FILTER, this);
        client.restrictTo(DOMAIN_FILTER);
        return client;
    }
}

File: app/src/main/java/me/devsaki/hentoid/activities/sources/BabeTodayActivity.kt
package me.devsaki.hentoid.activities.sources

import me.devsaki.hentoid.enums.Site

private const val DOMAIN_FILTER = "babe.today"
private val GALLERY_FILTER =
    arrayOf("https://babe.today/[%\\w\\-_/]+$", "https://m.babe.today/[%\\w\\-_/]+$")

class BabeTodayActivity : BaseBrowserActivity() {
    override fun getStartSite(): Site {
        return Site.BABETODAY
    }

    override fun allowMixedContent(): Boolean {
        return false
    }

    override fun createWebClient(): CustomWebViewClient {
        val client = CustomWebViewClient(getStartSite(), GALLERY_FILTER, this)
        client.restrictTo(DOMAIN_FILTER)
        return client
    }
}